### PR TITLE
Update to 'SubFlow State' spec doc

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tsurdilo@redhat.com>

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [ x] Specification
- [x ] Schema
- [ ] Examples
- [ ] Usecases
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:

Updates the "SubFlow" state doc section to make it more clear. 
Introduces the "standalone" top-level property that referenced workflow can use to indicate that they should only be 
invocable via other workflows.

### CODE REVIEW

**Comment by tsurdilo** [comment]
Related to Issue https://github.com/serverlessworkflow/specification/issues/140

**Comment by tsurdilo** [comment]
@jorgenj Do you mind taking a look? Does this make things more clear?

File: schema/workflow.json [hunk 1]
```diff
@@ -33,6 +33,11 @@
       "format": "uri",
       "description": "URI to JSON Schema that workflow data input adheres to"
     },
+    "standalone": {
+      "type": "boolean",
+      "default": true,
+      "description": "Indicates if the workflow can be executed directly or can be executed only from other workflows (via SubFlow state)"
```
[the remark this answers is not in the document]
Comment (jorgenj) [reply schema/workflow.json:39]: I have a question/suggestion that I just posted to #140 about this 'standalone' flag.

[the remark this answers is not in the document]
Comment (tsurdilo) [reply schema/workflow.json:39]: replied on the issue. i do like your idea of injection and it makes sense but I think it opens a door to hinter portability of the markup if runtimes have different merging / lookup criteria. Being sometimes a little more restrictive goes a long way in terms of portability :)

**Review by jorgenj — APPROVED** [review]
I really appreciate the updates regarding the question about 'waitForCompletion', I think that clears things up well for that question.

In my humble opinion, the standalone flag is something that could use some further discussion before adding.

**Review by ricardozanini — APPROVED** [review]

**Comment by tsurdilo** [comment]
@jorgenj after thinking about this I also think that function/event def inheritance is not something that is clear. I have removed the 'standalone' flag added here and added note that referenced workflows  (sub-flows) must declare their own function and event definitions.

**Comment by jorgenj** [comment]
Cool, I think that makes sense.

**Pull request merged**